crates/ruff_linter/src/rules/refurb/rules/slice_to_remove_prefix_or_suffix.rs

```diff
@@ -248,6 +248,30 @@ fn affix_removal_data<'a>(
         return None;
     }
     let slice = slice.as_slice_expr()?;
+
+    // Exit early if slice step is...
+    if slice
+        .step
+        .as_ref()
+        // present and
+        .is_some_and(|step| match step.as_ref() {
+            // not equal to 1
+            ast::Expr::NumberLiteral(ast::ExprNumberLiteral {
+                range: _,
+                value: ast::Number::Int(x),
+            }) => x.as_u8() != Some(1),
+            // and not equal to `None` or `True`
+            ast::Expr::NoneLiteral(_)
+            | ast::Expr::BooleanLiteral(ast::ExprBooleanLiteral {
+                range: _,
+                value: true,
+            }) => false,
```

Comment:
Nit: Use `..` so that this code doesn't need updating when we add a new field to `ExprNumberLiteral` or `BooleanLiteral`
```suggestion
            ast::Expr::NumberLiteral(ast::ExprNumberLiteral {
                value: ast::Number::Int(x),
                ..
            }) => x.as_u8() != Some(1),
            // and not equal to `None` or `True`
            ast::Expr::NoneLiteral(_)
            | ast::Expr::BooleanLiteral(ast::ExprBooleanLiteral {
                value: true,
                ..
            }) => false,
```

Reply:
Done!